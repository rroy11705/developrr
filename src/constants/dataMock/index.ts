import { PostInterface } from '../../interfaces/postInterface'

export const dataMockButton = {
  text: 'Any text'
}

export const dataMockErrorMessage = {
  text: 'Any text'
}

export const dataMockPostCard = {
  title: 'Any title',
  subtitle: 'Any subtitle',
  date: 'Any date',
  time: 'Any time',
  link: 'url.com',
  category: 'Any category',
  imagePath: '/image.com'
}

export const dataMockPostCardMini = {
  title: 'Any title',
  time: 'Any time',
  date: 'Any date',
  imagePath: '/image.com',
  link: 'url.com'
}

export const dataMockPosts: PostInterface[] = [
  {
    metadata: {
      category: 'Tech',
      cover: '/image.com',
      date: '21-04-2022',
      excerpt: 'This is a tech post',
      id: '1',
      link: '/posts/1',
      slug: 'post-1',
      time: '10 min',
      title: 'Tech Post 1'
    }
  },
  {
    metadata: {
      category: 'Science',
      cover: '/image.com',
      date: '21-04-2022',
      excerpt: 'This is a science post',
      id: '2',
      link: '/posts/2',
      slug: 'post-2',
      time: '15 min',
      title: 'Science Post 1'
    }
  },
  {
    metadata: {
      category: 'Science',
      cover: '/image.com',
      date: '21-04-2022',
      excerpt: 'This is another science post',
      id: '2',
      link: '/posts/3',
      slug: 'post-3',
      time: '15 min',
      title: 'Science Post 2'
    }
  }
]

export const mockHeaders = [
  {
    title: 'Science Post 1',
    description: 'This is a science post',
    keyword: 'science post',
    cover: '/image.com',
    slug: 'post-2',
    canonical: '/posts/2'
  }
]

export const dataMockSearchInPost = {
  text: 'Science Post',
  result: 2
}

export const dataMockSectionDoubleTitle = {
  titleMin: 'Any title min',
  titleMax: 'Any title max'
}

export const dataMockSectionTitle = {
  title: 'Any title',
  strong: false
}

export const dataMockCareer = {
  role: 'Any role',
  company: 'Any company',
  urlCompany: 'Any urlCompany',
  city: 'Any city',
  country: 'Any country',
  startDate: 'Any startDate',
  finalDate: 'Any finalDate',
  duration: 'Any duration'
}

export const dataMockFilter = {
  filters: ['Devops', 'Front end', 'SEO']
}

export const dataMockSocialCards = [
  {
    link: 'https://github.com/rroy11705',
    title: 'PORTFOLIO',
    subtitle: 'Last projects',
    icon: 'Any icon'
  },
  {
    link: 'https://twitter.com/DeveloprrPro',
    title: 'TWITTER',
    subtitle: 'Discussions',
    icon: 'Any icon'
  },
  {
    link: 'https://www.linkedin.com/in/rahulroy99/',
    title: 'LINKEDIN',
    subtitle: 'Connect to me',
    icon: 'Any icon'
  },
  {
    link: 'mailto:developrr.pro@gmail.com',
    title: 'EMAIL',
    subtitle: 'Talk to me',
    icon: 'Any icon'
  }
]
