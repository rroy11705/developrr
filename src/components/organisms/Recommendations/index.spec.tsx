// LatestPosts/index.test.js

import React from 'react'
import {
  render,
  screen,
  fireEvent
} from '../../../../test/config/react-library'
import LatestPosts from './index'
import { ThemeProvider } from 'styled-components'
import { dark } from '../../../../test/config/theme-wrapper'
import { dataMockPosts } from '../../../constants/dataMock/index'
import mockRouter from 'next-router-mock'

jest.mock('next/router', () => jest.requireActual('next-router-mock'))

jest.mock('../../../hooks/useWindowsize', () => () => ({
  width: 1200,
  height: 800
}))

const makeSut = posts => {
  return render(
    <ThemeProvider theme={dark}>
      <LatestPosts posts={posts} />
    </ThemeProvider>
  )
}

describe('LatestPosts component', () => {
  it('renders LatestPosts component with posts', () => {
    makeSut(dataMockPosts)

    const postCards = screen.queryAllByTestId('post-card')

    expect(postCards).toHaveLength(3)
  })
  it('renders explore button', () => {
    makeSut(dataMockPosts)

    const exploreIcon = screen.getByTestId('explore')

    expect(exploreIcon).toBeInTheDocument()
  })
  it('redirects to explore page on clicking explore', () => {
    makeSut(dataMockPosts)

    const exploreIcon = screen.getByTestId('explore')
    fireEvent.click(exploreIcon)

    expect(mockRouter).toMatchObject({
      asPath: '/explore',
      pathname: '/explore',
      query: {}
    })
  })
  it('renders correctly with window size', () => {
    const { container } = makeSut(dataMockPosts)

    expect(container).toMatchSnapshot()
  })
})
