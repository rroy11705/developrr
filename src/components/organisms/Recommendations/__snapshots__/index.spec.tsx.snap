// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LatestPosts component renders correctly with window size 1`] = `
<div>
  <div
    class="sc-aXZVg cMBBmU"
  >
    <div
      class="sc-fqkvVR hLTSNh"
      data-testid="double-title"
    >
      <div
        class="titleMin"
      >
        Latest posts
      </div>
      <div
        class="titleMax"
      >
        Written
      </div>
    </div>
    <div
      class="postCardContainer"
    >
      <div
        class="card"
      >
        <div
          class="sc-gEvEer dxlyna"
          data-testid="post-card"
        >
          <a
            href="/post-2"
          >
            <div
              class="cardContainer"
            >
              <div
                class="cardImage"
              >
                <div
                  class="tag"
                >
                  <div
                    class="sc-eqUAAy qmycC"
                    data-testid="category"
                  >
                    Science
                  </div>
                </div>
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 117.64705882352942% 0px 0px 0px;"
                  />
                  <img
                    alt="Main card image"
                    class="cardImageIn"
                    data-nimg="responsive"
                    data-testid="image"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimage.com&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimage.com&w=640&q=75 640w, /_next/image?url=%2Fimage.com&w=750&q=75 750w, /_next/image?url=%2Fimage.com&w=828&q=75 828w, /_next/image?url=%2Fimage.com&w=1080&q=75 1080w, /_next/image?url=%2Fimage.com&w=1200&q=75 1200w, /_next/image?url=%2Fimage.com&w=1920&q=75 1920w, /_next/image?url=%2Fimage.com&w=2048&q=75 2048w, /_next/image?url=%2Fimage.com&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
              <div
                class="cardDescription"
              >
                <div
                  class="content"
                >
                  <div
                    class="title"
                  >
                    Science Post 1
                  </div>
                  <div
                    class="subtitle"
                  >
                    This is a science post
                  </div>
                </div>
                <div
                  class="details"
                >
                  <div
                    class="time"
                  >
                    <svg
                      class="clockIcon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                      />
                    </svg>
                     
                    15 min
                  </div>
                  <div
                    class="date"
                  >
                    21-04-2022
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="sc-gEvEer dxlyna"
          data-testid="post-card"
        >
          <a
            href="/post-3"
          >
            <div
              class="cardContainer"
            >
              <div
                class="cardImage"
              >
                <div
                  class="tag"
                >
                  <div
                    class="sc-eqUAAy qmycC"
                    data-testid="category"
                  >
                    Science
                  </div>
                </div>
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 117.64705882352942% 0px 0px 0px;"
                  />
                  <img
                    alt="Main card image"
                    class="cardImageIn"
                    data-nimg="responsive"
                    data-testid="image"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimage.com&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimage.com&w=640&q=75 640w, /_next/image?url=%2Fimage.com&w=750&q=75 750w, /_next/image?url=%2Fimage.com&w=828&q=75 828w, /_next/image?url=%2Fimage.com&w=1080&q=75 1080w, /_next/image?url=%2Fimage.com&w=1200&q=75 1200w, /_next/image?url=%2Fimage.com&w=1920&q=75 1920w, /_next/image?url=%2Fimage.com&w=2048&q=75 2048w, /_next/image?url=%2Fimage.com&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
              <div
                class="cardDescription"
              >
                <div
                  class="content"
                >
                  <div
                    class="title"
                  >
                    Science Post 2
                  </div>
                  <div
                    class="subtitle"
                  >
                    This is another science post
                  </div>
                </div>
                <div
                  class="details"
                >
                  <div
                    class="time"
                  >
                    <svg
                      class="clockIcon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                      />
                    </svg>
                     
                    15 min
                  </div>
                  <div
                    class="date"
                  >
                    21-04-2022
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="sc-gEvEer dxlyna"
          data-testid="post-card"
        >
          <a
            href="/post-1"
          >
            <div
              class="cardContainer"
            >
              <div
                class="cardImage"
              >
                <div
                  class="tag"
                >
                  <div
                    class="sc-eqUAAy qmycC"
                    data-testid="category"
                  >
                    Tech
                  </div>
                </div>
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 117.64705882352942% 0px 0px 0px;"
                  />
                  <img
                    alt="Main card image"
                    class="cardImageIn"
                    data-nimg="responsive"
                    data-testid="image"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimage.com&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimage.com&w=640&q=75 640w, /_next/image?url=%2Fimage.com&w=750&q=75 750w, /_next/image?url=%2Fimage.com&w=828&q=75 828w, /_next/image?url=%2Fimage.com&w=1080&q=75 1080w, /_next/image?url=%2Fimage.com&w=1200&q=75 1200w, /_next/image?url=%2Fimage.com&w=1920&q=75 1920w, /_next/image?url=%2Fimage.com&w=2048&q=75 2048w, /_next/image?url=%2Fimage.com&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
              <div
                class="cardDescription"
              >
                <div
                  class="content"
                >
                  <div
                    class="title"
                  >
                    Tech Post 1
                  </div>
                  <div
                    class="subtitle"
                  >
                    This is a tech post
                  </div>
                </div>
                <div
                  class="details"
                >
                  <div
                    class="time"
                  >
                    <svg
                      class="clockIcon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                      />
                    </svg>
                     
                    10 min
                  </div>
                  <div
                    class="date"
                  >
                    21-04-2022
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="card explore"
        data-testid="explore"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
